### ISSUE TITLE
MD013: Incorrect line length calculation with `reflow = true` and link in list item

### ISSUE THREAD

**Issue by sisp** [rvben/rumdl#466]
### rumdl version

0.1.28

### Installation method

Pre-built binary

### Bug description

I've stumbled over a line length calculation issue when `reflow = true` and a link definition exists in multi-paragraph list item, which is valid Markdown. When the length of the link definition crosses some threshold, the `MD013` rule is violated with a too high length. The reported line length seems to include the length of the preceding paragraph because it changes as the preceding paragraph length changes.

### Steps to reproduce

1. Create a file `test.md` with this content:

    ```markdown
    # Test MD013

    - Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod

        [a link]

        [a link]: https://localhost/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    ```

1. Run `rumdl check test.md`.

1. See the error below.

### Expected behavior

```console
$ rumdl check test.md

Success: No issues found in 1 file (7ms)
```

### Actual behavior

```console
$ rumdl check test.md
test.md:3:1: [MD013] Line length 160 exceeds 80 characters [*]

Issues: Found 1 issues in 1 file (31ms)
Run `rumdl fmt` to automatically fix 1 of the 1 issues
```

Note that the reported line length is 160 characters which is clearing incorrect.

### Configuration

```toml
[global]
line-length = 80

[MD013]
reflow = true
```

### Operating system

Linux

### Additional context

1. Edit `test.md` and shorten the target link in the link definition by one character:

    ```diff
     # Test MD013

     - Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod

         [a link]

    -    [a link]: https://localhost/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    +    [a link]: https://localhost/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    ```

1. Run `rumdl check test.md` and observe no error:

    ```console
    $ rumdl check test.md

    Success: No issues found in 1 file (7ms)
    ```

    Note that _only one_ character was removed, so the reported line length of 160 seems to have dropped to at most 80. 🤔

When `reflow = false`, there is no linter error. And when the link definition is outside/below the list, there is no linter error even for very long links.

**Comment by sisp** [comment]
It seems the problem also occurs with inline links, not only with link definitions.

With this Markdown content

```markdown
# Test MD013

- Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod

    [a link](https://localhost/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx)
```

there's an error:

```console
$ rumdl check test.md
test.md:3:1: [MD013] Line length 151 exceeds 80 characters [*]

Issues: Found 1 issues in 1 file (7ms)
Run `rumdl fmt` to automatically fix 1 of the 1 issues
```

Note the reported line length of 151 characters, it appears this is the full unwrapped line:

```markdown
- Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod [a link](https://localhost/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx)
```

When removing one character from the link URL, there's no error anymore.

**Comment by rvben** [comment]
Fixed in v0.1.30. The reflow logic was incorrectly joining link reference definitions into the paragraph length calculation. List item paragraphs are now checked independently, and link ref defs are exempt from reflow. Thanks for the clear repro steps, @sisp!

**Comment by sisp** [comment]
Thank you very much for the super quick fix, @rvben! :bow:
